src/events/message-search.ts

```diff
@@ -0,0 +1,207 @@
+) as ParserRules;
+const parser = parserFor(ourRules);
+// Can improve in future to do the entire processing with this parser to just grab the search tokens we want
+
+// https://discord.com/developers/docs/reference#message-formatting-formats
+const mentionPatterns = (
+	["UserWithOptionalNickname", "Channel", "Role", "SlashCommand", "Emoji", "Timestamp"] as const
+).map(key => new RegExp(FormattingPatterns[key], "g"));
+
+export function cleanMessageMarkup(message: string): string {
+	// Remove the above markup elements
+	const nodes = parser(message, { inline: true });
+	message = nodes
+		.filter(node => node.type === "text")
+		.map(node => node.content)
+		.join("");
+	for (const regex of mentionPatterns) {
+		message = message.replaceAll(regex, "");
+	}
+	return message;
+}
+
+const DELIMITERS = {
+	ANGLE: { match: /<([^<\n]+?)>/g, prune: /<<.*?>>/g },
+	SQUARE: { match: /\[([^<\n]+?)\]/g, prune: /\[\[.*?\]\]/g },
+	CC: { match: /=([^<\n]+?)=/g, prune: /==.*?==/g },
+	TRANS: { match: /\)([^<\n]+?)\(/g, prune: null }, // ) (
+	HAVEN: { match: /%([^<\n]+?)\^/g, prune: null } // % ^
+} as const;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function getDelimiter(message: Message) {
+	switch (message.guildId) {
+		case "170669983079071745":
+			return DELIMITERS.CC;
+		case "597478047163219988":
+		case "845400480452050954":
+			return DELIMITERS.SQUARE;
+		case "871582695967301703":
+			return DELIMITERS.HAVEN;
+		case "780952232103116800":
+			return DELIMITERS.TRANS;
+		default:
+			return DELIMITERS.ANGLE;
+	}
+}
+
+export function parseSummons(cleanMessage: string, regex: RegExp): string[] {
+	return [...cleanMessage.matchAll(regex)]
+		.map(match => match[1].trim())
+		.filter(summon => {
+			// Ignore matches containing only whitespace or containing the following three tokens
+			if (summon.length === 0) {
+				return false;
+			}
+			const lower = summon.toLowerCase();
+			return !["://", "(", "anime"].some(token => lower.includes(token));
+		});
+}
+
+export function preprocess(
+	message: string,
+	delimiter: typeof DELIMITERS[keyof typeof DELIMITERS] = DELIMITERS.ANGLE
+): string[] {
+	message = cleanMessageMarkup(message);
+	if (delimiter.prune) {
+		message = message.replaceAll(delimiter.prune, "");
+	}
+	return parseSummons(message, delimiter.match);
+}
+
+/**
+ * Capture groups:
+ * - kid: % sign, if this matches a Konami ID, optional
+ * - number: matched sequence of digits
+ * - lang: one of the locales,optional
+ *
+ * %4007
+ * %4007,en
+ * % 4007  ,en
+ * 00010000
+ * 00010000,ja
+ */
+const NUMERIC_REGEX = new RegExp(
+	/^(?<kid>%)?\s*(?<number>\d+)\s*(?:,(?<lang>LOCALES))?$/
+		.toString()
+		.slice(1, -1)
+		.replace("LOCALES", LOCALES.join("|"))
+);
+
+/**
+ * Capture groups:
+ * - text: main body
+ * - inputLang: one of the locales, optional
+ * - resultLang: one of the locales, optional, if exists then inputLang exists
+ *
+ * blue-eyes
+ * baguette,fr
```

Comment:
I was going to be a bit 🤨 at this choice of example, but it _does_ appear in French card names, e.g. Wonder Wand (Baguette Magique).

Reply:
It actually just means rod or stick in French, despite the English meaning to only refer to the bread 😅 
I think it's a recognizable example of French that also actually shows up in cards, usually wands in English though like Wonder Wand and World Legacy - "World Wand" (Héritage du Monde - "Baguette du Monde")